packages/javascript-extension/src/index.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { IRenderMime } from '@jupyterlab/rendermime-interfaces';
+import { RenderedJavaScript } from '@jupyterlab/rendermime';
+
+export const TEXT_JAVASCRIPT_MIMETYPE = 'text/javascript';
+export const APPLICATION_JAVASCRIPT_MIMETYPE = 'application/javascript';
+
+function evalInContext(code: string, element: Element, document: Document, window: Window) {
+   // tslint:disable-next-line
+  return eval(code);
+}
+
+export class ExperimentalRenderedJavascript extends RenderedJavaScript {
+  render(model: IRenderMime.IMimeModel): Promise<void> {
+    const renderJavascript = () => {
+      try {
+        const data = model.data[this.mimeType] as string;
+        evalInContext(data, this.node, document, window);
```

Comment:
Here you eval the code before trust is checked. I would move the block checking trust to the top of this function.

Reply:
Oh, yes, I see that now..